pdfminer/pdftypes.py

```diff
@@ -185,6 +186,23 @@ def stream_value(x):
     return x
 
 
+def decompress_corrupted(data):
+    d = zlib.decompressobj(zlib.MAX_WBITS | 32)
+    f = io.BytesIO(data)
+    result_str = b''
+    buffer = f.read(1)
+    i = 0
+    try:
+        while buffer:
+            result_str += d.decompress(buffer)
+            buffer = f.read(1)
+            i += 1
+    except zlib.error:
+        if i < len(data) - 3:
+            raise
```

Comment:
This `3` seems to be cherry-picked specifically for the PDF in case. Maybe just always ignore the error if not in `settings.STRICT`mode?

Reply:
Well it has been empirically found using this file and another one :) The idea is that the error fixed here comes from some CRC check at the end of the data, which has been found to be in the last 4 bytes. This could probably be confirmed by the zlib spec. https://docs.python.org/3/library/zlib.html#zlib.adler32 is about int32 checksum, which seems consistent.

If we just plain raise the error in STRICT mode, the corrupted file won't be readable in this mode as before this patch. This may be fine but could be even done before calling `decompress_corrupted`

Will a some comment about this in the code.

Reply:
If it is in `settings.STRICT` mode it should raise the error asap and not try the `decompress_corrupted()` at all. That's how things are right now, so no changes needed there. 

Since `decompress_corrupted` only deals with the non strict mode, I think it's fair to just extract as much as possible and not raise the error ever. 

But I can also see your argument that when it raises an error before the last 4 bytes there is data missing. And we should make the user aware of that. Maybe raise a warning in that case? See pdfpage.py:140 for an example on how to do that.

Reply:
done

Reply:
> If it is in settings.STRICT mode it should raise the error asap and not try the decompress_corrupted() at all. 

It is still trying `decompress_corrupted()` in strict mode. That should not happen.